src/lib.rs

```diff
@@ -164,7 +164,7 @@ impl UPTransportZenoh {
             uri.authority_name.clone()
         };
         // ue_id
-        let ue_id = if uri.has_wildcard_entity_id() {
+        let ue_id = if uri.has_wildcard_entity_type() {
```

Comment:
I think you'd want to check for both `uri.has_wildcard_entity_type()` and `uri.has_wildcard_entity_instance()` as this will be checking if both the entity type and instance are wildcarded before making the ue_id wildcarded.

Reply:
I suppose this might mean the tests and examples will need to also get updated to comply with this.

Reply:
FMPOV the key needs to be `*` if the entity type is `0xFFFF` to make sure that we match messages from/to arbitrary services. In this case the instance ID is irrelevant.

However, if just the instance ID is `0xFFFF`, then we would still need to make sure to _only_ match messages to/from the specific entity type.

Example: For an entity type of `0x1234` and instance type `0xFFFF` (wildcard), the key's entity fragment should thus probably _not_ just be `*` (which would match all entity types), but something like `*1234`.

@evshary is this possible and would this also match `1234`, i.e. an entity type with the default (omitted) instance id (`0000`)?

Reply:
Hmm, I see what you mean @sophokles73. What do you say @evshary?

Reply:
Okay, I understand what you mean, @sophokles73. To achieve what you describe above, we need to use `$*` instead of `*` in Zenoh. I've added the changes.
@PLeVasseur I also added some tests. Please let me know whether this is what you expect.